### ISSUE TITLE
Validation layer crashes in vkQueueSubmit if submit info contains timeline semaphore info, but also uses binary semaphores

### ISSUE THREAD

**Issue by aqnuep** [KhronosGroup/Vulkan-ValidationLayers#2763]
**Describe the Issue**
Consider the following scenario:
`vkQueueSubmit()` waits on binary semaphore, signals timeline semaphore.

Thus `VkSubmitInfo` has the following field values:
```
waitSemaphoreCount = 1
pWaitSemaphores = { <binary_semaphore> }
signalSemaphoreCount = 1
pSignalSemaphores = { <timeline_semaphore> }
```

... and the chained `VkTimelineSemaphoreSubmitInfo` has the following field values:
```
waitSemaphoreValueCount = 0
pWaitSemaphoreValues = NULL
signalSemaphoreValueCount = 1
pSignalSemaphoreValues = { <timeline_semaphore_value> }
```

This causes a crash in the validation layer in `ValidationStateTracker::PostCallRecordQueueSubmit()` at the following line:
```
uint64_t value = timeline_semaphore_submit ? timeline_semaphore_submit->pWaitSemaphoreValues[i] : 0;
```

The validation layer incorrectly expects `pWaitSemaphoreValues` to be non-NULL in this case, as per spec we have the following valid usages:

> VUID-VkSubmitInfo-pNext-03240
> If the pNext chain of this structure includes a VkTimelineSemaphoreSubmitInfo structure and any element of pWaitSemaphores was created with a VkSemaphoreType of VK_SEMAPHORE_TYPE_TIMELINE, then its waitSemaphoreValueCount member must equal waitSemaphoreCount

> VUID-VkTimelineSemaphoreSubmitInfo-pWaitSemaphoreValues-parameter
> If waitSemaphoreValueCount is not 0, and pWaitSemaphoreValues is not NULL, pWaitSemaphoreValues must be a valid pointer to an array of waitSemaphoreValueCount uint64_t values

However, neither applies here, as none of `pWaitSemaphores`'s elements is a timeline semaphore (only those of `pSignalSemaphores`), thus `waitSemaphoreValueCount` can be `0` and `pWaitSemaphoreValues` can be NULL, yet the validation layer code expects otherwise.

In fact, the validation layer assumes just by the presence of the `VkTimelineSemaphoreSubmitInfo` structure in the chain of `VkSubmitInfo` to be an indicator that it has a non-NULL `pWaitSemaphoreValues` field, which is a much stricter requirement than those defined by the spec.

**Valid Usage ID**
Please check applicable valid usage cases of `VkSubmitInfo` and `VkTimelineSemaphoreSubmitInfo`.

**Environment:**
 - OS: Any
 - GPU: Any
 - SDK version: 1.2.170
 - Options enabled (synchronization, best practices, etc.): Standard Preset

**Comment by ncesario-lunarg** [comment]
Thanks for the detailed information @aqnuep! Looks like this is possibly addressed in #2620. This will be included in the upcoming 1.2.176 SDK. If building the layers from source is an option for you, you should be able to try the fix out now.

Please re-open this issue if the problem still exists on the latest source.

**Comment by aqnuep** [comment]
Thanks, indeed PR #2620 should get this issue fixed.
I apologize for the redudant issue report, I should have searched also in the already closed issues when checking whether this was reported already and then I would have found #2616.
